apollo-runtime/src/main/java/com/apollographql/android/impl/RealApolloCall.java

```diff
@@ -132,17 +157,53 @@ private RealApolloCall(Operation operation, HttpUrl serverUrl, Call.Factory http
   }
 
   @Override @Nonnull public ApolloCall<T> clone() {
-    return new RealApolloCall<>(operation, serverUrl, httpCallFactory, httpCache, moshi, responseBodyConverter, cache,
-        cacheControl);
+    return new RealApolloCall<>(operation, serverUrl, httpCallFactory, httpCache, cacheControl, moshi,
+        responseFieldMapper, customTypeAdapters, cache);
+  }
+
+  private Response<T> cachedResponse() {
+    if (cacheControl == CacheControl.NETWORK_ONLY) {
+      return null;
+    }
+
+    T cachedData = cachedData();
+    if (cachedData != null) {
+      return new Response<>(operation, cachedData, null);
+    }
+
+    if (cacheControl == CacheControl.CACHE_ONLY) {
+      return new Response<>(operation, null, null);
```

Comment:
Kotlin I guess because it will bring runtime with it, here we use pure java. But I know this is would be nicer to have on Kotlin. Maybe we need to consider some `Optional` approach instead of null